debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java

```diff
@@ -47,6 +50,13 @@ protected GCClassArrayClassSlotIterator(J9ClassPointer clazz) throws CorruptData
 			slots.add(slot);
 			addresses.add(VoidPointer.cast(clazz.arrayClassEA()));
 		}
+		if (valueTypeHelper.isJ9ClassAValueType(clazz)) {
```

Comment:
If the clazz is an array, this check will be false. Isn't the arrayClass->companionArray slot missed ?